libs/checksum/src/main/java/org/codice/ddf/checksum/impl/Sha256ChecksumProvider.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.checksum.impl;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import org.apache.commons.codec.binary.Hex;
+import org.codice.ddf.checksum.AbstractChecksumProvider;
+
+public class Sha256ChecksumProvider extends AbstractChecksumProvider {
+
+  private static final String DIGEST_ALGORITHM = "SHA-256";
+
+  @Override
+  public String calculateChecksum(InputStream inputStream)
+      throws IOException, NoSuchAlgorithmException {
+    if (inputStream == null) {
+      throw new IllegalArgumentException("InputStream cannot be null");
+    }
+
+    try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+      byte[] buffer = new byte[1028];
+
+      int len;
+      while ((len = inputStream.read(buffer)) != -1) {
+        outputStream.write(buffer, 0, len);
```

Comment:
Okay. I think I see the difference here. The Adler flavor is using a special pass-through stream that handles checksum computation. That's not a built-in option for SHA-256 so we need to do a manual handoff between streams. 

The local byte array `buffer` is only being used to cache the handoff. The `outputStream` has an internal byte array that will resize itself as necessary. 

I think this will work fine. 

(❓ ) Do we need to handle the case where `len` is zero?

Reply:
Found a different utility that simplifies this